### DISCUSSION TITLE
Dynamically turn flare on

### DISCUSSION THREAD

**Discussion by Jellyfrog** [spatie/laravel-ignition#106] [Ideas]
We have the need of dynamically turning flare on based on user prefs,
if we try to set `flare.key`, in for example our AppServiceProvider, it won't work since the config is only read on package boot here;
https://github.com/spatie/laravel-ignition/blob/5bd36b98e54fdabe02c0738164a8e2494b11fc32/src/IgnitionServiceProvider.php#L117-L122

We can ofc put it in "dont-discover" and boot at a later stage, but that feels very hackish, and it wouldnt allow to turn off flare either (since the key would be cached)

**Comment by freekmurze** [comment]
I would get the instance from the container, call `setApiToken` on that instance, and then set that instance in the container (I think the `App` has a `swap` method that allows you to do this.

**Comment by Jellyfrog** [comment]
Thanks that could work. Went the other way instead, manually loading it
